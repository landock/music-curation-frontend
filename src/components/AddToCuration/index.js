import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
class AddToCuration extends PureComponent {
  render() {
    const markup = this.props.curations ? (
      this.props.curations.map((curation, index) => (
        <div key={index}>
          <input value={curation} type="radio" name="curations" />
          <label>{curation}</label>
        </div>
      ))
    ) : (
      <p>No curations available</p>
    );
    return (
      <div>
        <span>Add to:</span>
        <div>
          <input type="search" placeholder="Search" />
        </div>
        {markup}
      </div>
    );
  }
}

AddToCuration.propTypes = {
  curations: PropTypes.arrayOf(PropTypes.string),
};

export default AddToCuration;
