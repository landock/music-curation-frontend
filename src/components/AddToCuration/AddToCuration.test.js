
import React from 'react'
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import AddToCuration from '.'

it('AddToCuration: default', () => {
  const component = renderer.create(<AddToCuration />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})