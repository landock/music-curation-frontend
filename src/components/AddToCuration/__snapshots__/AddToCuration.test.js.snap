// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCuration: default 1`] = `
<div>
  <span>
    Add to:
  </span>
  <div>
    <input
      placeholder="Search"
      type="search"
    />
  </div>
  <p>
    No curations available
  </p>
</div>
`;
